### PR TITLE
[FIX] tests: fix iot freezegun dependency

### PR DESCRIPTION
**Pull request by Ysoroko**
On the IoT Box we do not install `freezegun` pip package. This leads to an error with Odoo failing to start on the IoT Box when connected to a database in master.
This PR fixes the issue by wrapping the import of freezetime in a try/except block and only using the module when it's imported properly

Related PR which introduced the issue: https://github.com/odoo/odoo/pull/176944

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qle-odoo** [comment]
not sure to be needed
@robodoo delegate+

File: odoo/tests/common.py [hunk 1]
```diff
@@ -2162,11 +2166,14 @@ def __call__(self, func):
             func.freeze_time = self.time_to_freeze
             return func
         else:
-            return freezegun.freeze_time(self.time_to_freeze)(func)
+            if freezegun:
```
Comment (Xavier-Do) [inline odoo/tests/common.py:2169]: I would add a warning here telling that freezegun is missing (and in __enter__)

**Comment by d-fence** [comment]
> There is no reason for odoo.tests.common to be imported in order to load the iotbox in production. That's what you need to fix.
> 
> Most likely it's because either the iot or one of its dependencies imports its `tests` from the root package, there's a bunch doing that:
> 
> * [b946693](https://github.com/odoo/odoo/commit/b946693df5ee7275412265152fc7f5dc3a872ef1)
> * [odoo/enterprise@b18a1ac](https://github.com/odoo/enterprise/commit/b18a1ac438254e56d4011dcfcbe66d7110fc747e)

I think that it will be addressed in #177671. So adding, just a warning as xdo said is probably the way to go until that.

**Comment by Ysoroko** [comment]
@xmo-odoo @d-fence if you consider this is unnecesary and the mentionned PR will fix the issue for everyone I can close the PR
I added logger warnings suggested by @Xavier-Do so we can merge it otherwise

**Pull request closed without merging**